runtime/bcutil/ComparingCursor.cpp

```diff
@@ -416,12 +428,20 @@ ComparingCursor::shouldCheckForEquality(DataType dataType, U_32 u32Value)
 		return false;
 	}
 
+	if(isLambda()){
+		/* if the class is a lambda class we compare everything except the class name */
+		if(SRP_TO_UTF8_CLASS_NAME == dataType)
+			return false;
```

Comment:
Can you try move this to Line 444. We skip SRP_TO_UTF8_CLASS_NAME if isLambda() is true. For all other types, we return the same as the non-lambda class.